opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/RsaJwtsVerifyingServiceImpl.java

```diff
@@ -84,6 +82,24 @@ public boolean verify(String signature, String encodedPublicKey, ConfirmConsentD
         ));
     }
 
+    @Override
+    public boolean verify(String signature, String encodedPublicKey, PaymentInitiationDataToSign data) {
+        return createDataToSign(signature, encodedPublicKey, data.getBankId(), data.getFintechUserId(), data.getRedirectOk(), data.getRedirectNok(), data.getXRequestId(), data.getInstant(),
+                                data.getOperationType());
+    }
+
+    private boolean createDataToSign(String signature, String encodedPublicKey, String bankId, String fintechUserId, String redirectOk, String redirectNok, UUID xRequestId, Instant instant,
```

Comment:
Looks like body is mostly duplicate to
https://github.com/adorsys/open-banking-gateway/pull/738/files#diff-a4d480f039ef3160a509a35345f1ce31R103

maybe we could extract:
```
Map<String, String> values = new HashMap<>();
        values.put(HttpHeaders.BANK_ID, bankId);
        values.put(HttpHeaders.FINTECH_USER_ID, fintechUserId);
        values.put(HttpHeaders.FINTECH_REDIRECT_URL_OK, redirectOk);
        values.put(HttpHeaders.FINTECH_REDIRECT_URL_NOK, redirectNok);
        DataToSign dataToSign = new DataToSign(xRequestId, instant, operationType, values);
```
to have DataToSignFactory or something similar

Reply:
DataToSignFactory  will be implemented as a part of https://github.com/adorsys/open-banking-gateway/pull/773 (the reason is that this pull request is about tests,  and there will be implemented complete data signing logic)